addons/purchase_stock/models/stock_move.py

```diff
@@ -246,7 +246,12 @@ def _get_valuation_price_and_qty(self, related_aml, to_curr):
 
     def _is_purchase_return(self):
         self.ensure_one()
-        return self.location_dest_id.usage == "supplier"
+        return self.location_dest_id.usage == "supplier" or (self.location_dest_id in self._return_transit_locations())
+
+    @api.model
+    def _return_transit_locations(self):
+        res = self.env.ref('stock.stock_location_inter_wh')
```

Comment:
no need to create a function, use the xml_id `_is_purchase_return()` 
and don't forget the `raise_if_not_found` key